### ISSUE TITLE
v0.11.0: Model directory creation and loading error

### ISSUE THREAD

**Issue by RobGeada** [kserve/modelmesh-runtime-adapter#41]
Model loading fails with the v0.11 images where the model uses a HTTP protocol storageUri:

Error message:
`Failed to create model directory and load model {"modelId":  $model_name, "error": "Error calling stat on /models/$model_name__isvc-xxx/. : start /models/$model_name/... not a directory"`

Downgrading to the v0.10 image eliminates the problem.

**Comment by ckadner** [comment]
do you have an examples model / ISVC and how you deployed it -- I'm curious why it says `$model_name` in the error message

**Comment by ckadner** [comment]
Thanks Rob. And you deployed a ISVC like this?

```YAML
apiVersion: serving.kserve.io/v1beta1
kind: InferenceService
metadata:
  name: example-sklearn-isvc
  annotations:
    serving.kserve.io/deploymentMode: ModelMesh
spec:
  predictor:
    model:
      modelFormat:
        name: sklearn
      runtime: mlserver-0.x
      storageUri: "https://github.com/trustyai-explainability/trustyai-explainability/raw/main/explainability-service/demo/models/model.joblib?raw=true"
```

And the same `storageUri` worked in `v0.10.0`?

**Comment by tjohnson31415** [comment]
I was able to reproduce the error and the investigation has surfaced some interesting (buggy) behavior:

The HTTP downloader in the puller will actually download the pointed-at model to a filename that matches the internal generated directory name instead of creating a file inside the directory. This happens because the `modelPath` sent by the controller when using an HTTP `storageURI` ISVC is an empty string, but the puller code is written assuming a non-empty `modelPath`. So the result of the pull is a file structure like:
```
$ cat /models/xgboost-mushroom-http__isvc-574ed0b433 | head -n5
{
  "learner": {
    "attributes": {},
    "gradient_booster": {
      "model": {
...
```

With the [changes to support PVC](https://github.com/kserve/modelmesh-runtime-adapter/pull/36), code was added that also assumes a non-empty model path and has:
```
	modelFullPath := modelDir + string(filepath.Separator) + modelPathFilename
```
[REF](https://github.com/kserve/modelmesh-runtime-adapter/pull/36/files#diff-d2a332b0b59e3a7eb5342fbac94a7579fb5141229326af37664a3b20c1308fa1R192)

For the HTTP case, `modelPathFilename` is empty, so `modelFullPath` becomes `/models/xgboost-mushroom-http__isvc-574ed0b433/` (with a trailing `/` even though `xgboost-mushroom-http__isvc-574ed0b433` is a file, not directory). When that path is [passed into `os.Stat` in the adapter code](https://github.com/kserve/modelmesh-runtime-adapter/blob/f9dc1dca42eae9491e941398127e3ae55d9cfaf2/model-mesh-mlserver-adapter/server/server.go#L173-L176), it results in the `Error calling stat` error reported in this bug report.

I'm work on getting some PRs pushed up to improve this situation:
1. change how HTTP models are handled so that model data is written into the generated directory instead of replacing it
1. use `filepath.Join()` instead of string concatenation with a separator since it handles empty elements (we can't go back to SecureJoin because that breaks PVC support)
1. better handling of an empty model path in the puller code more generally (maybe just returning an error in that case)
